os-server/src/main/java/app/openschool/user/UserController.java

```diff
@@ -53,9 +62,42 @@ public ResponseEntity<Set<PreferredCategoryDto>> savePreferredCategories(
   }
 
   @GetMapping("/{userId}/courses/enrolled")
-  @Operation(summary = "find user's courses by course status")
-  public ResponseEntity<List<UserCourseDto>> findUserEnrolledCourses(
+  @Operation(
+      summary = "find user's courses by course status",
+      security = @SecurityRequirement(name = "bearerAuth"))
+  public ResponseEntity<List<UserCourseDto>> findEnrolledCourses(
       @PathVariable Long userId, @RequestParam(required = false) Long courseStatusId) {
     return ResponseEntity.ok(this.userService.findUserEnrolledCourses(userId, courseStatusId));
   }
+
+  @GetMapping("/{userId}/courses/saved")
+  @Operation(
+      summary = "find user's saved courses",
+      security = @SecurityRequirement(name = "bearerAuth"))
+  public ResponseEntity<?> findSavedCourses(
+      Pageable pageable, @PathVariable Long userId, Locale locale) {
+    if (userService.findById(userId).isEmpty()) {
+      return ResponseEntity.badRequest()
+          .body(new ResponseMessage(messageSource.getMessage("user.not.exists", null, locale)));
+    }
+    return ResponseEntity.ok(this.userService.findUserSavedCourses(pageable, userId));
+  }
+
+  @PostMapping("/{userId}/courses/{courseId}/saved")
+  @Operation(summary = "save course", security = @SecurityRequirement(name = "bearerAuth"))
+  public ResponseEntity<ResponseMessage> saveCourse(
+      @PathVariable Long userId, @PathVariable Long courseId, Locale locale) {
+    if (userService.findById(userId).isEmpty()) {
+      return ResponseEntity.badRequest()
+          .body(new ResponseMessage(messageSource.getMessage("user.not.exists", null, locale)));
+    }
+    if (courseService.findById(courseId).isEmpty()) {
+      return ResponseEntity.badRequest()
+          .body(new ResponseMessage(messageSource.getMessage("course.not.exists", null, locale)));
+    }
+    this.userService.saveCourse(userId, courseId);
+    return ResponseEntity.ok(
```

Comment:
just return saved course, or its id, it is actually obvious that course saved when we return 200 ok, the message part can be handled from the frontend part